guidelines/configuration-api-design.md

```diff
@@ -0,0 +1,36 @@
+## Designing configuration API's
+
+All configuration API's should be code first since that allows us to best guide the users and allows us to evolve the API's using obsolete messages.
+
+Configuration API's should be variable based and not returning `this`.
+
+Example:
+
+```
+var myFeatureConfig = endpointConfig.EnableMyFeature()
```

Comment:
I think this example is misleading since it could be confused with 

endpointConfig.EnableFeature<MyFeature>()